core/src/test/scala/unit/kafka/server/metadata/BrokerMetadataListenerTest.scala

```diff
@@ -248,28 +254,28 @@ class BrokerMetadataListenerTest {
       assertEquals(200L, listener.highestMetadataOffset)
 
       // Check that we generate at least one snapshot once we see enough records.
-      assertEquals(-1L, snapshotter.prevCommittedOffset)
-      generateManyRecords(listener, 1000L)
-      assertEquals(1000L, snapshotter.prevCommittedOffset)
-      assertEquals(1000L, snapshotter.activeSnapshotOffset)
-      snapshotter.activeSnapshotOffset = -1L
+      assertNull(snapshotter.prevSnapshotId)
+      generateManyRecords(listener, endOffset = 1000L)
+      assertEquals(new OffsetAndEpoch(1000L, 0), snapshotter.prevSnapshotId)
+      assertEquals(new OffsetAndEpoch(1000L, 0), snapshotter.activeSnapshotId)
+      snapshotter.activeSnapshotId = null
 
       // Test creating a new snapshot after publishing it.
       val publisher = new MockMetadataPublisher()
       listener.startPublishing(publisher).get()
-      generateManyRecords(listener, 2000L)
+      generateManyRecords(listener, endOffset = 2000L)
       listener.getImageRecords().get()
-      assertEquals(2000L, snapshotter.activeSnapshotOffset)
-      assertEquals(2000L, snapshotter.prevCommittedOffset)
+      assertEquals(new OffsetAndEpoch(2000L, 0), snapshotter.prevSnapshotId)
+      assertEquals(new OffsetAndEpoch(2000L, 0), snapshotter.activeSnapshotId)
 
       // Test how we handle the snapshotter returning false.
-      generateManyRecords(listener, 3000L)
-      assertEquals(2000L, snapshotter.activeSnapshotOffset)
-      generateManyRecords(listener, 4000L)
-      assertEquals(2000L, snapshotter.activeSnapshotOffset)
-      snapshotter.activeSnapshotOffset = -1L
-      generateManyRecords(listener, 5000L)
-      assertEquals(5000L, snapshotter.activeSnapshotOffset)
+      generateManyRecords(listener, endOffset = 3000L)
+      assertEquals(new OffsetAndEpoch(2000L, 0), snapshotter.activeSnapshotId)
+      generateManyRecords(listener, endOffset = 4000L)
+      assertEquals(new OffsetAndEpoch(2000L, 0), snapshotter.activeSnapshotId)
+      snapshotter.activeSnapshotId = null
+      generateManyRecords(listener, endOffset = 5000L)
+      assertEquals(new OffsetAndEpoch(5000L, 0), snapshotter.activeSnapshotId)
```

Comment:
Given my comments regarding `mockBatchReader`. I would expect these snapshot id comparisons to fail, no?